__tests__/api.test.js

```diff
@@ -188,3 +211,61 @@ describe('error handling', () => {
     })
 })
 
+describe('error handling', () => {
+test('responds with a 400 status code when given an invalid perameter', () => {
+    const newComment = {
+        username: 'butter_bridge',
+        body: 'This is a new comment'
+    }
+    return request(app)
+    .post('/api/articles/string/comments')
+    .send(newComment)
+    .expect(400)
+    .then(res => {
+        const { msg } = res.body;
+        expect(msg).toBe('Bad request');
+    })
+})
+test('responds with a 404 status code when given an invalid article id', () => {
+    const newComment = {
+        username: 'butter_bridge',
+        body: 'This is a new comment'
+    }
+    return request(app)
+    .post('/api/articles/999999/comments')
+    .send(newComment)
+    .expect(404)
+    .then(res => {
+        const { msg } = res.body;
+        expect(msg).toBe('Article not found');
+    })
+})
+test('responds with a 404 status code when given an invalid username', () => {
+    const newComment = {
+        username: 'invalid_username',
+        body: 'This is a new comment'
+    }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(newComment)
+    .expect(404)
+    .then(res => {
+        const { msg } = res.body;
+        expect(msg).toBe('User not found');
+    })
+})
+test('responds with a 400 status code when given an invalid body', () => {
+    const newComment = {
+        username: 'butter_bridge',
+        body: ''
+    }
+    return request(app)
+    .post('/api/articles/1/comments')
+    .send(newComment)
+    .expect(400)
+    .then(res => {
+        const { msg } = res.body;
+        expect(msg).toBe('Both username and body are required');
+    })
+})
```

Comment:
We need a sad path test for when username or body properties are missing entirely please